frontend/webapp/reuseable-components/tooltip/index.tsx

```diff
@@ -27,17 +26,17 @@ const TooltipContainer = styled.div`
 export const Tooltip: React.FC<TooltipProps> = ({ text, withIcon, children }) => {
   const [isHovered, setIsHovered] = useState(false);
   const [popupPosition, setPopupPosition] = useState<Position>({ top: 0, left: 0 });
+  const popupRef = useRef<HTMLDivElement>(null);
 
-  const handleMouseEvent = (e: React.MouseEvent) => {
+  const handleMouseEvent = (e: MouseEvent) => {
     const { type, clientX, clientY } = e;
     const { innerWidth, innerHeight } = window;
 
     let top = clientY;
     let left = clientX;
-    const textLen = text?.length || 0;
 
-    if (top >= innerHeight / 2) top += -40;
-    if (left >= innerWidth / 2) left += -(textLen * 8);
+    if (top >= innerHeight / 2) top += -(popupRef.current?.clientHeight || 40);
+    if (left >= innerWidth / 2) left += -(popupRef.current?.clientWidth || Math.min((text?.length || 0) * 7.5, 300));
```

Comment:
It seems a little bit risky, did you try this with long text?

Reply:
Yes I did, long text was the reason this was applied